.devcontainer/configure.py

```diff
@@ -24,11 +23,9 @@
     "nvidia": "nvidia-bleeding-edge",
 }
 
-
-def run_shell(cmd):
-  cmd = shlex.split(cmd)
-  return subprocess.run(cmd, check=True, text=True,
-                        stdout=subprocess.PIPE).stdout
+configure_base = SourceFileLoader("configure_base",
+                                  str(CURRENT_DIR.parent /
+                                      "configure.py")).load_module()
```

Comment:
It looks like this is creating a circular dependency between the root level configure script and this configure script? I'd also love to avoid having to do this path thing in a novel way each time. Can we instead do this how some of our python build tools are already doing it? That is, put common utilities under `build_tools/python` and add that directory to the path at the beginning of each script like this:

https://github.com/openxla/iree/blob/0c0c34f7c8d5a920942f888db4521f64737d598c/build_tools/benchmarks/benchmark_helper.py#L9-L13

That gives us something that we can mostly copy-paste in every script. Then we can just do "normal" imports.

Unfortunately it still depends on where the file is in the directory tree, but generalizing that makes the code block we'd be copy-pasting bigger. We can code-golf it later, but let's at least have one way to do this.

So I would put whatever shared code in `build_tools/python/configure/`, ideally with useful file names and then use it here and the top-level configure.py